khmer/_khmermodule.cc

```diff
@@ -912,9 +912,10 @@ hash_consume(khmer_KCountingHash_Object * me, PyObject * args)
                         "string length must >= the hashtable k-mer size");
         return NULL;
     }
-
+    std::string read(long_str);
+    bool is_valid;
     unsigned int n_consumed;
-    n_consumed = counting->consume_string(long_str);
+    n_consumed = counting->check_and_process_read(read, is_valid);
```

Comment:
On Mon, May 18, 2015 at 08:08:32AM -0700, Tamer Mansour wrote:
> >      unsigned int n_consumed;
> > -    n_consumed = counting->consume_string(long_str);
> > +    n_consumed = counting->check_and_process_read(read, is_valid);
> 
> It should not break anything. I did not fix consume_string(). I used another function instead which do the check then call consume_string. I am testing the new functionally by trying data file with 'N's characters.

Yes, but I want a test that demonstrates that something has changed about
the functionality.  This test should break when consume_string is used and
succeed when check_and_process_read is used.